mlrun/datastore/utils.py

```diff
@@ -180,3 +181,53 @@ def get_kafka_brokers_from_dict(options: dict, pop=False) -> typing.Optional[str
             FutureWarning,
         )
     return kafka_bootstrap_servers
+
+
+def transform_list_filters_to_tuple(additional_filters):
+    tuple_filters = []
+    if not additional_filters:
+        return tuple_filters
+    validate_additional_filters(additional_filters=additional_filters)
+    for additional_filter in additional_filters:
+        tuple_filters.append(tuple(additional_filter))
+    return tuple_filters
+
+
+def validate_additional_filters(additional_filters):
+    nan_error_message = "using NaN in additional_filters is not supported"
+
+    if not additional_filters:
+        return
+    for filter_tuple in additional_filters:
+        if not filter_tuple:
+            continue
```

Comment:
Probably not good enough? Sure, you might have an empty list there for some reason, but you could also have other stuff that just happens to evaluate to boolean false, e.g. zero, empty string, `None`...

Reply:
nice catch!

Reply:
+ added unit test case